### PR TITLE
chore(sub): various UI tweaks

### PR DESCRIPTION
**Pull request by genehynson**
Closes https://github.com/influxdata/data-acquisition/issues/487
Closes https://github.com/influxdata/data-acquisition/issues/495
Closes https://github.com/influxdata/data-acquisition/issues/481
Closes https://github.com/influxdata/data-acquisition/issues/499
Closes https://github.com/influxdata/data-acquisition/issues/462

## Fixes error message notifications

Previously showed no text in the alert. Now:

Screen Shot 2022-08-12 at 1 59 22 PM

## Timestamp precision names

Previously showed the abbreviation (e.g. 'NS'). Now:

image

But still uses abbreviation in the request:
Screen Shot 2022-08-12 at 2 00 13 PM

## Protocol scheme bug fix:

Used to show `TCP://mqtt://[host]:[port]`. Now:

image

## Remove hostname scheme (if provided)

The user should not include their own scheme in the hostname - we now auto-remove it if they do:

https://user-images.githubusercontent.com/6411855/184446775-6547c7c0-13dc-45b0-93b1-05c2593c90eb.mov

## Subscription create/update/delete events:

Create: `subscription creation failure` / `subscription creation successful` / `subscription creation attempted`
Update: `subscription update failure` / `subscription update successful` / `subscription update attempted`
Delete: `subscription delete failure` / `subscription delete successful` / `subscription delete attempted`

## Max field lengths

Added max field lengths to match db schema. If no limit is enforced in the db I set it to 1024 in the UI.

## Json / Avro name character enforcement

Only chars [A-Za-z0-9_] are allowed for measurement/field/tags for Json data format

image

image

image

## Port validation

Validates that the provided port is a number and is in the valid port range. 

image

 
### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] Feature flagged, if applicable

### CODE REVIEW

File: src/writeData/subscriptions/components/SubscriptionFormContent.tsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ const SubscriptionFormContent: FC<Props> = ({
                   )
                 }
                 status={edit ? status : ComponentStatus.Disabled}
-                maxLength={56}
+                maxLength={255}
```
Comment (genehynson) [inline src/writeData/subscriptions/components/SubscriptionFormContent.tsx:88]: I'm not sure why this was `56` before - was that intentional?

Comment (helenosheaa) [reply src/writeData/subscriptions/components/SubscriptionFormContent.tsx:88]: no idea, I think it's an age old bug

**Review by helenosheaa — APPROVED** [review]
lgtm, thanks for attacking these bugs!

**Review by subirjolly — APPROVED** [review]

File: src/writeData/subscriptions/utils/form.ts [hunk 2]
```diff
@@ -412,3 +417,32 @@ const parseDate = (timeString: string) => {
 
   return parsed.getTime()
 }
+
+export const handlePortValidation = (port: any) => {
+  try {
```
[the remark this answers is not in the document]
Comment (genehynson) [reply src/writeData/subscriptions/utils/form.ts:422]: Nice catch - let me fix

**Review by subirjolly — APPROVED** [review]

**Pull request merged**